src/main/dashboard/app.tsx

```diff
@@ -127,57 +134,73 @@ function LoaderForm() {
             />
           </Match>
           <Match when={loaderType() === 'File'}>
             <input
               type="file"
               accept=".js"
               class={styles.input}
-              placeholder={formatMessage({id: 'eureka.loader.file.placeholder', default: 'Choose a file'})}
+              onChange={(e) => {
+                const files = e.currentTarget.files;
+                if (files && files.length > 0) {
+                  const file = files[0];
+                  setFile(file); // Update signal with the selected file
+
+                  // Read the file as text and store its content
+                  const reader = new FileReader();
+                  reader.onload = () => {
+                    setFileContent(reader.result as string); // Store the file content
+                  };
+                  reader.readAsText(file); // Read the file as text
+                } else {
+                  setFile(null); // Reset if no file is selected
+                  setFileContent(null); // Reset file content if no file is selected
+                }
+              }}
             />
           </Match>
         </Switch>
 
-        <button
-          class={styles.button}
-          disabled={shouldDisable()}
+        {/* Button Element */}
+        <button 
+          class={styles.button} 
+          disabled={shouldDisable()} 
           onClick={async () => {
-          switch (loaderType()) {
-            case 'URL':
-              setLoading(true);
-              await eureka.load(extensionURL());
-              setLoading(false);
-              break;
-            case 'Code':
-              setLoading(true);
-              await eureka.load(stringToDataURL(extensionCode()));
-              setLoading(false);
-              break;
-            case 'File':
-              if (extensionFile()) {
-                const reader = new FileReader();
-                reader.onload = async () => {
-                  setLoading(true);
-                  eureka.load(reader.result as string);
-                  setLoading(false);
-                };
-                reader.readAsText(extensionFile());
+            setLoading(true); // Set loading to true at the start of loading process
+            try {
+              switch (loaderType()) {
+                case 'URL':
+                  await eureka.load(extensionURL());
+                  break;
+                case 'Code':
+                  await eureka.load(stringToDataURL(extensionCode()));
+                  break;
+                case 'File':
+                  if (fileContent()) { // Use stored file content
+                    await eureka.load(stringToDataURL(fileContent()));
+                  }
+                  break;
               }
-              break;
-          }
-        }}>
+            } catch (error) {
+              console.error("Error loading extension:", error); // Handle any errors that occur during loading
```

Comment:
May we should show error below the "Load" button?

Reply:
I added it. I also fixed a bug where projects do not load when the project is saved without eureka

Reply:
Should I go ahead and resolve this? Or do you have anything else to add?